packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +648,843 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+  // e.g.: change is on the end of the tag => change will be added to the tag if it's not mention and after the tag if it's mention
+  let processedChange = change;
+  // end tag plain text index of the last processed tag
+  let lastProcessedPlainTextTagEndIndex = 0;
+  // as some tags/text can be removed fully, selectionEnd should be updated correctly
+  let changeNewEndIndex: number | null = null;
+
+  for (let i = 0; i < tags.length; i++) {
+    const tag = tags[i];
+    if (tag.plainTextBeginIndex === undefined) {
+      continue;
+    }
+    // all plain text indexes includes trigger length for the mention that shouldn't be included in
+    // htmlText.substring because html strings don't include the trigger
+    // mentionTagLength will be set only for 'msft-mention', otherwise should be 0
+    let mentionTagLength = 0;
+    let isMentionTag = false;
+    if (tag.tagType === 'msft-mention') {
+      mentionTagLength = mentionTrigger.length;
+      isMentionTag = true;
+    }
+
+    //change start is before the open tag
+    if (startIndex <= tag.plainTextBeginIndex) {
+      // Math.max(lastProcessedPlainTextTagEndIndex, startIndex) is used as startIndex may not be in [[previous tag].plainTextEndIndex - tag.plainTextBeginIndex] range
+      const startChangeDiff = tag.plainTextBeginIndex - Math.max(lastProcessedPlainTextTagEndIndex, startIndex);
+
+      result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx - startChangeDiff) + processedChange;
+      console.log('updateHTML 0 result after update', result);
+      if (oldPlainTextEndIndex <= tag.plainTextBeginIndex) {
+        //the whole change is before tag start
+        // oldPlainTextEndIndex already includes mentionTag length
+        const endChangeDiff = tag.plainTextBeginIndex - oldPlainTextEndIndex;
+        lastProcessedHTMLIndex = tag.openTagIdx - endChangeDiff;
+        processedChange = '';
+        // the change is handled; exit
+        break;
+      } else {
+        // change continues in the tag
+        lastProcessedHTMLIndex = tag.openTagIdx;
+        processedChange = '';
+        // proceed to the next check
+      }
+    }
+    let plainTextEndIndex = 0;
+    let closeTagIdx = 0;
+    let closeTagLength = 0;
+    if (tag.plainTextEndIndex !== undefined && tag.closeTagIdx !== undefined) {
+      // close tag exists
+      plainTextEndIndex = tag.plainTextEndIndex;
+      closeTagIdx = tag.closeTagIdx;
+      // tag.tagType.length + </>
+      closeTagLength = tag.tagType.length + 3;
+    } else {
+      //no close tag
+      plainTextEndIndex = tag.plainTextBeginIndex;
+      closeTagIdx = tag.openTagIdx + tag.openTagBody.length;
+      closeTagLength = 0;
+    }
+
+    if (startIndex < plainTextEndIndex) {
+      // change started before the end tag
+      if (startIndex <= tag.plainTextBeginIndex && oldPlainTextEndIndex === plainTextEndIndex) {
+        console.log('updateHTML 1.1 result', result);
```

Comment:
remove

Reply:
All of them will be removed in the next PRs for this feature as this is WIP right now

Reply:
Removed